src/math/Euler.js

```diff
@@ -243,6 +243,101 @@ class Euler {
 
 	}
 
+	// Based on: Bernardes E, Viollet S (2022) Quaternion to Euler angles conversion:
+	// A direct, general and computationally efficient method. PLoS ONE 17(11): e0276302.
+	setFromQuaternionDirect( q, order, update ) {
+
+		let half_sum;
+		let half_diff;
+
+		switch ( order ) {
+
+			case 'XYZ':
+				this._y = Math.acos( clamp( - 2 * q.w * q.y - 2 * q.x * q.z, - 1, 1 ) );
+				half_sum = - Math.atan2( - q.x - q.z, q.w + q.y );
+				half_diff = - Math.atan2( - q.x + q.z, q.w - q.y );
+				break;
+
+			case 'XZY':
+				this._y = Math.acos( clamp( - 2 * q.w * q.z + 2 * q.x * q.y, - 1, 1 ) );
```

Comment:
Does is make sense that `.y` should get special treatment regardless of the Euler order?

Reply:
Yep, `.y` is easier to calculate in the sense that it is always well-defined, for a very simple reason: In Proper symmetric Euler angles, asymmetric Tait-Bryan angles or even generalized Davenport angles, the second axis is **always orthogonal** to the other two (meaning that there is no `ZZY` order, for example).

`.x` and `.z`, on the other hand, are not always well-defined, depending on the value of `.y`

For Tait-Bryan / asymmetric sequences: 
- When `.y = -pi/2`, the first axis is aligned with the third axis, the system becomes underdetermined and only `.x + .y` is well-defined.
- When `.y = +pi/2`, the first axis is aligned with the third axis, the system becomes underdetermined and only `.x - .y` is well-defined.
- In every other case, both `.x + .y` and `.x - .y` are well-defined.

Reply:
`.y` is not the "second axis" in three.js. The ordering is specified by `Euler.order`.

Reply:
Ok so, maybe I misunderstood the definition here. I thought `.x`, `.y` and `.z` were just attributes for the first, second and third angles in the `Euler` instance (at least that is what would make more sense to me for a generalized approach with Proper angles and Tait-Bryan angles).

In this case, for example, in the order `YXZ`, `.y` would be a rotation in `.x`.

Do you mean to say that, for example, the value in `.y` is always a rotation around the axis `Y`? (That would probably explain why  symmetric and asymmetric axes are used in three.js).

Reply:
>.y is always a rotation around the axis Y?

yes.